🪟 🎨 Refining BulkEditPanel component

YatsukBogdan1 opened the pull request:
## What
Closes #20372 
Closes #20731 


## How
- Updates rendering logic, for the cases while Primary Key or Cursor is not available or defined by source
- We also had discussion with @edmundito that `Switch` styling will be done in separate PR

## Screenshots
<img width="1219" alt="Screenshot 2022-12-14 at 16 24 30" src="https://user-images.githubusercontent.com/25852567/207621688-91b280a8-0822-4b93-bd94-1af90ccdb964.png">
<img width="1141" alt="Screenshot 2022-12-14 at 16 24 37" src="https://user-images.githubusercontent.com/25852567/207621693-da26681d-aedc-4281-803d-227bd03be77c.png">

![Screenshot 2022-12-22 at 15 18 27](https://user-images.githubusercontent.com/25852567/209142964-ab53b5cd-9868-4543-9c2d-76a23414d8c2.png)
![Screenshot 2022-12-22 at 15 18 31](https://user-images.githubusercontent.com/25852567/209142966-161b658f-c7ba-479c-8ce4-67cd5040e0b7.png)
![Screenshot 2022-12-22 at 15 18 48](https://user-images.githubusercontent.com/25852567/209142967-80130e9a-0753-4ffd-928f-12926036fe2b.png)

tealjulia commented:
> the cases while Primary Key or Cursor is not available

Would that be in cases where all of the selected streams do not have at least one column with the same name?  Are there any other cases?

krishnaglick commented:
Failing snapshot tests :)

krishnaglick commented on airbyte-webapp/src/components/connection/CatalogTree/next/StreamPathSelect.tsx:
```diff
@@ -37,19 +35,18 @@ type PathPopoutProps = StreamPathSelectBaseProps & (StreamPathSelectMultiProps |
-      );
+      node = props.isMulti ? props.path.map(pathDisplayName).join(", ") : pathDisplayName(props.path);
+    } else {
+      node = <FormattedMessage id="connection.catalogTree.sourceDefined" />;
```
Nit: This could be in a `useMemo`

krishnaglick reviewed (commented):
Should this PR have more tests added?

YatsukBogdan1 commented:
> > the cases while Primary Key or Cursor is not available
> 
> Would that be in cases where all of the selected streams do not have at least one column with the same name? Are there any other cases?
@tealjulia for now we decided to put only design updates in this PR, and keep any logic updates in separate one

> Should this PR have more tests added?
@krishnaglick as we discussed on the stand up yesterday, the component itself is dumb, and just renders what is passed from `useBulkEditService` hook, so the component itself is tested, and lets keep any tests related to the logic, into another PR

edmundito reviewed (changes requested):
I re-tested again and did notice a discrepancy in the table itself. It looks like the non-selectable values have a background, but that only applies to the bulk edit

Here you can see that id is not selectable but it has the pill background around it
![Screen Shot 2022-12-21 at 16 23 58](https://user-images.githubusercontent.com/168664/209005277-fa8fecc2-eaca-413f-9047-4714f7cbccc3.png)

But this is really similar to what's needed in #20731

YatsukBogdan1 commented:
@edmundito I updated table row cursor and primary key as you suggested

![Screenshot 2022-12-22 at 15 18 27](https://user-images.githubusercontent.com/25852567/209142868-d619be22-b924-4e31-b741-4660fbbb090a.png)
![Screenshot 2022-12-22 at 15 18 31](https://user-images.githubusercontent.com/25852567/209142870-2eb6b3ee-f8a2-4885-8c39-fb8d1f6e3e15.png)
![Screenshot 2022-12-22 at 15 18 48](https://user-images.githubusercontent.com/25852567/209142872-f9379eaa-a4a3-4e14-a4ac-2b474f9255b8.png)

YatsukBogdan1 commented:
Added a few more updates:
- removed animation for `<BulkEditPanel />`
- added `<StreamPathSelect />` new property `pillIfChangeUnavailable` for cases when we need or not to render pill under unavailable to change path text
<img width="1150" alt="image" src="https://user-images.githubusercontent.com/25852567/209211500-3fcfb2bd-bad1-45b6-a165-1d1612a0e0f3.png">

edmundito commented:
@YatsukBogdan1 Just checked the latest changes and I noticed that what I see doesn't match the screenshot int he previous comment.
![Screen Shot 2023-01-03 at 14 39 05](https://user-images.githubusercontent.com/168664/210429286-92cc062c-3ace-4d17-a3a0-ff6b8815531b.png)

YatsukBogdan1 commented:
updated PR with next changes:
- disables pointer events for `PillButton` component while `disabled: true`

edmundito commented on airbyte-webapp/src/components/connection/CatalogTree/next/StreamPathSelect.tsx:
```diff
@@ -19,6 +18,10 @@ interface StreamPathSelectBaseProps {
+  // This property is used for cases when the path is defined by source, therefore
+  // in some cases we need this path to render with pill background (BulkEditPanel) and
+  // in some cases it should be only text (StreamsTable)
+  pillIfChangeUnavailable?: boolean;
```
Maybe `withSourceDefinedPill` is more clear?

edmundito commented on airbyte-webapp/src/components/ui/PillSelect/PillSelect.tsx:
```diff
@@ -34,7 +36,7 @@ export const PillSelect: React.FC<PillSelectProps> = ({ className, ...props }) =
                 className={className}
               >
-                {label}
+                {disabled ? <FormattedMessage id="connectionForm.bulkEdit.pillButtonLabel.notAvailable" /> : label}
```
This formatted message is part of the connection form, but this component is part of the UI, so it crosses different domains. We're not sure if the default behavior for disabled is always to say "not available"

YatsukBogdan1 replied:
@edmundito I added `renderDisabledState: () => React.ReactNode` property for `PillSelect`, so now, when `disabled: true`, the content can be controlled from different application parts. please let me know if this looks good to you

edmundito reviewed (changes requested):
Tested locally and the functionality looks good but had some comments about the implmentation.

edmundito reviewed (commented):
LGTM. Tested locally.
